src/include/types.h

```diff
@@ -489,10 +489,13 @@ struct errorcode32_t {
   // cppcheck-suppress noExplicitConstructor
   errorcode32_t(int32_t i) : code(i) {}
 
-  operator int() const { return code; }
-  int operator==(int i) {
-    return code==i;
-  }
+  operator int() const  { return code; }
+  int* operator&()      { return &code; }
+  int operator==(int i) { return code==i; }
+  int operator>(int i)  { return code>i; }
+  int operator>=(int i) { return code>=i; }
+  int operator<(int i)  { return code>i; }
```

Comment:
http://pulpito.ceph.com/kchai-2017-07-05_01:57:37-rados-wip-kefu-testing2-distro-basic-smithi/

should be `code < i`.